system/handlers/formcontrols/DerivativePicker.cfc

```diff
@@ -0,0 +1,34 @@
+component output=false {
+
+	property name="assetManagerService"  inject="AssetManagerService";
+
+	public string function index( event, rc, prc, args={} ) output=false {
+
+		var derivatives = assetManagerService.listEditorDerivates();
+		var args.labels = [];
+		var args.values = [];
+		var args.width  = [];
+		var args.height = [];
+
+		for( var derivative in derivatives.value ) {
+		    args.values.append( derivative );
+			args.labels.append( translateResource( uri="derivatives:#derivative#.title", defaultValue="derivatives:#derivative#.title" ) );
+		}
+
+		for( var transformation in derivatives.dimension ) {
```

Comment:
This is not quite right - a derivative can be more than just a resize - it could potentially add a watermark and do other things. Instead of having the dimensions inserted into the image URL, we need the name of the derivative and nothing more. (this note applies to other areas of the code).